repository/src/main/java/org/alfresco/service/cmr/rule/RuleService.java

```diff
@@ -235,8 +235,8 @@
      * @param rule Rule
      */
     @Auditable(parameters = {"nodeRef", "rule"})
-    public void saveRule(NodeRef nodeRef, Rule rule);
-    
+    public Rule saveRule(NodeRef nodeRef, Rule rule);

```

Comment:
I don't think we should change this existing (almost sine beginning of the world) public contract. 
Maybe adding a new method which would do the trick here?

Reply:
Yes, new method and deprecate old. Given that RuleService is part of "cmr" package. We could also check whether explicitly marked as @PublicApi (embedded Java level) or not.

Reply:
It's not part of the public API, so I felt it was fair game.  I can add a new method instead if that's preferred.

Reply:
Actually I'm not going to bother about this. I'm only changing the return from `void` so:
* Anyone calling the method won't notice a difference
* Anyone extending/implementing the interface will need to update their code regardless of the approach.

Reply:
> If it stays and a new method is added, you don't need to do anything (however, you may want to).
In fact it's almost certainly true that the only case you need to look at this new method is if you've implemented the existing `saveRules` method.  See as an example the AGS implementation - if I hadn't seen the compilation error then I wouldn't have realised I needed to update the implementation.

I think this is an argument for _not_ keeping the existing method.